biorxiv/word_vector_experiment/biorxiv_generate_embeddings.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Generate BioRxiv Document Embeddings 
+
+# This notebook is designed to generate document embeddings for every article in bioRxiv.
+
+# In[2]:
+
+
+from pathlib import Path
+import os
+
+from gensim.models import Word2Vec
+from gensim.parsing.preprocessing import remove_stopwords
+import lxml.etree as ET
+import pandas as pd
+from sklearn.manifold import TSNE
+from tqdm import tqdm_notebook
+import umap
+
+
+# # Output Documents to File
+
+# This section dumps all of biorxiv text into a single document in order to train the word2vec model. This is for ease of training the model.
```

Comment:
I think we discussed this a little, but just to clarify.  Your training set consists of all sentences from all the documents together.  So you're not keeping track of sentences from the same document?  So you're loosing some context about where the sentences came from and the meaning of words can differ depending on the overall context of the article..Or does this give a more robust representation of the words because they are trained over many different contexts...?

Reply:
> Your training set consists of all sentences from all the documents together. So you're not keeping track of sentences from the same document? 

Correct. I am training the word2vec model using all sentences in **every** biorxiv document. 

>So you're loosing some context about where the sentences came from and the meaning of words can differ depending on the overall context of the article..Or does this give a more robust representation of the words because they are trained over many different contexts...?

The latter is what is happening. Since I am using every biorxiv document, I can avoid biases such as spurious associations or missing associations. (e.g. two words are associated in a neruoscience context but would have been missed if I stuck with a bioinformatics context.)